### PR TITLE
feat(cloud.logs): add kibana endpoints

### PR DESCRIPTION
**Pull request by pdepaepe**
Hello,

This PR will add support to new Kibana endpoints available on dbaas/logs namespace.

Thanks for your time on this subject.

Signed-off-by: Pierre De Paepe <pierre.de-paepe@ovhcloud.com>

### CODE REVIEW

File: src/api/dbaas/logs/kibana/kibana.aapi.service.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        kibana.resetCache();
+    };
+  
+    kibana.resetCache = function () {
+        cache.removeAll();
+    };
+  
+    return kibana;
+  });
+
```
Comment (antleblanc) [inline src/api/dbaas/logs/kibana/kibana.aapi.service.js:26]: Extra empty line (please refer to the EditorConfig file.

File: src/api/dbaas/logs/logs.service.js [hunk 2]
```diff
@@ -53,6 +53,9 @@ angular.module('ovh-api-services').service('OvhApiDbaasLogs', ($injector) => ({
   Dashboard() {
     return $injector.get('OvhApiDbaasLogsDashboard');
   },
+  Kibana() {
```
[the remark this answers is not in the document]
Comment (pdepaepe) [reply src/api/dbaas/logs/logs.service.js:56]: No, it's the same for Alias() and Index()

File: src/api/dbaas/logs/kibana/kibana.v6.service.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+angular.module('ovh-api-services').service('OvhApiDbaasLogsKibanaV6', ($resource, $cacheFactory) => {
```
[the remark this answers is not in the document]
Comment (pdepaepe) [reply src/api/dbaas/logs/kibana/kibana.v6.service.js:1]: Do i have to rename Alias and Index too, move them all in new folders ?

File: src/api/dbaas/logs/kibana/kibana.v6.service.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      },
+    };
+  
+    const kibanaResource = $resource('/dbaas/logs/:serviceName/output/elasticsearch/kibana/:kibanaId', {
+      serviceName: '@serviceName',
+      kibanaId: '@kibanaId',
+    }, {
+      query: { method: 'GET', isArray: true, cache: queryCache },
+      create: { method: 'POST', interceptor },
+      remove: { method: 'DELETE', interceptor },
```
[the remark this answers is not in the document]
Comment (pdepaepe) [reply src/api/dbaas/logs/kibana/kibana.v6.service.js:18]: Not required as, 2API fetch it.

File: src/api/dbaas/logs/kibana/kibana.v6.service.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      cache.removeAll();
+    };
+  
+    kibanaResource.resetQueryCache = function () {
+      queryCache.removeAll();
+    };
+  
+    return kibanaResource;
+  });
+
```
[the remark this answers is not in the document]
Comment (pdepaepe) [reply src/api/dbaas/logs/kibana/kibana.v6.service.js]: Fixed.

File: src/api/dbaas/logs/kibana/kibana.service.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+      return $injector.get('OvhApiDbaasLogsKibanaV6');
+    },
+    Iceberg() {
+      return $injector.get('OvhApiDbaasLogsKibanaIceberg');
+    },
+    Aapi() {
+      return $injector.get('OvhApiDbaasLogsKibanaAapi');
+    },
+  }));
+
```
[the remark this answers is not in the document]
Comment (pdepaepe) [reply src/api/dbaas/logs/kibana/kibana.service.js]: Fixed.

File: src/api/dbaas/logs/kibana/kibana.iceberg.service.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    const kibanaResource = iceberg('/dbaas/logs/:serviceName/output/elasticsearch/kibana/:kibanaId', {
+      serviceName: '@serviceName',
+      kibanaId: '@kibanaId',
+    }, {
+      create: { method: 'POST' },
+    });
+  
+    return kibanaResource;
+  });
+
```
[the remark this answers is not in the document]
Comment (pdepaepe) [reply src/api/dbaas/logs/kibana/kibana.iceberg.service.js]: Fixed.

File: src/api/dbaas/logs/kibana/kibana.aapi.service.js (hunk 1 above)
Comment (pdepaepe) [reply src/api/dbaas/logs/kibana/kibana.aapi.service.js]: Fixed.

File: src/api/dbaas/logs/logs.service.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (antleblanc) [reply src/api/dbaas/logs/logs.service.js:56]: Adding new Services should be aligned with what's the documentation expect and mention.
You can refer to: https://github.com/ovh-ux/ovh-api-services#structure

As you mentioned, both `Alias` and `Index` services aren't located into the right location and we have to reorganize them but it means that we will integrate a Breaking Change at this point, so I would recommend you to start adding new Services into the right location and then we will handle the move for other Services later on.

File: src/api/dbaas/logs/kibana/kibana.v6.service.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (antleblanc) [reply src/api/dbaas/logs/kibana/kibana.v6.service.js:1]: Renaming them will introduce a Breaking Change, so you can just start adding new Kibana services into the right location and we will handle the move in a separate Pull Request.

File: src/api/dbaas/logs/logs.service.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (pdepaepe) [reply src/api/dbaas/logs/logs.service.js]: Move to output/elasticsearch/kibana. Could you please tell me if it's acceptable like this.

[the remark this answers is not in the document]
Comment (antleblanc) [reply src/api/dbaas/logs/logs.service.js]: Looks good to me except that some files are not well formatted based on the [EditorConfig rules](https://github.com/ovh-ux/ovh-api-services/blob/master/.editorconfig).

Could you please take a look?

Thanks!

File: src/api/dbaas/logs/output/elasticsearch/elasticsearch.service.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+angular.module('ovh-api-services').service('OvhApiDbaasLogsOutputElasticsearchKibana', ($injector) => ({
+    Kibana() {
+      return $injector.get('OvhApiDbaasLogsOutputElasticsearch');
+    },
+  }));
+
```
Comment (antleblanc) [inline src/api/dbaas/logs/output/elasticsearch/elasticsearch.service.js:6]: Could you please apply the EditConfig rules?

File: src/api/dbaas/logs/output/elasticsearch/kibana/kibana.aapi.service.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    kibana.resetAllCache = function () {
+        kibana.resetCache();
+    };
+  
+    kibana.resetCache = function () {
+        cache.removeAll();
+    };
+  
+    return kibana;
+  });
```
Comment (antleblanc) [inline src/api/dbaas/logs/output/elasticsearch/kibana/kibana.aapi.service.js:25]: Could you please apply the EditConfig rules?

File: src/api/dbaas/logs/output/elasticsearch/kibana/kibana.iceberg.service.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+angular.module('ovh-api-services').service('OvhApiDbaasLogsOutputElasticsearchKibanaIceberg', (iceberg) => {
+    const kibanaResource = iceberg('/dbaas/logs/:serviceName/output/elasticsearch/kibana/:kibanaId', {
+      serviceName: '@serviceName',
+      kibanaId: '@kibanaId',
+    }, {
+      create: { method: 'POST' },
+    });
+  
+    return kibanaResource;
+  });
```
Comment (antleblanc) [inline src/api/dbaas/logs/output/elasticsearch/kibana/kibana.iceberg.service.js:10]: Could you please apply the EditConfig rules?

File: src/api/dbaas/logs/output/elasticsearch/kibana/kibana.service.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+    v6() {
+      return $injector.get('OvhApiDbaasLogsOutputElasticsearchKibanaV6');
+    },
+    Iceberg() {
+      return $injector.get('OvhApiDbaasLogsOutputElasticsearchKibanaIceberg');
+    },
+    Aapi() {
+      return $injector.get('OvhApiDbaasLogsOutputElasticsearchKibanaAapi');
+    },
+  }));
```
Comment (antleblanc) [inline src/api/dbaas/logs/output/elasticsearch/kibana/kibana.service.js:11]: Could you please apply the EditConfig rules?

File: src/api/dbaas/logs/output/elasticsearch/kibana/kibana.v6.service.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+    kibanaResource.resetCache = function () {
+      cache.removeAll();
+    };
+  
+    kibanaResource.resetQueryCache = function () {
+      queryCache.removeAll();
+    };
+  
+    return kibanaResource;
+  });
```
Comment (antleblanc) [inline src/api/dbaas/logs/output/elasticsearch/kibana/kibana.v6.service.js:35]: Could you please apply the EditConfig rules?

File: src/api/dbaas/logs/output/output.service.js [hunk 13]
```diff
@@ -0,0 +1,6 @@
+angular.module('ovh-api-services').service('OvhApiDbaasLogsOutput', ($injector) => ({
+    Elasticsearch() {
+      return $injector.get('OvhApiDbaasLogsOutputElasticsearch');
+    },
+  }));
+
```
Comment (antleblanc) [inline src/api/dbaas/logs/output/output.service.js:6]: Could you please apply the EditConfig rules?

File: src/api/dbaas/logs/logs.service.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (pdepaepe) [reply src/api/dbaas/logs/logs.service.js]: Might be better now.

**Comment by pdepaepe** [comment]
> Do you have an ETA when these new Endpoints will be available? (Still flagged as Internal)

Endpoints are about to be GA.

**Pull request merged**